cogs/twitcasting/twitcasting.py

```diff
@@ -0,0 +1,190 @@
+        self.connection.delete_record(channel_id=channel_id, user_id=user_id)
+        self.twitcasting.remove_webhook(user_id=user_id)
+        await ctx.send(f"Deleted user {user_id} from list of webhooks.")
+
+    @tc.command()
+    async def rename(self, ctx, webhook_name: str):
+        """
+        Migrates subscriptions to a new webhook with the provided name. Deletes the previous webhook.
+            Usage: z!tc rename <new-webhook-name>
+        """
+        channel_id = f'{ctx.guild.id}:{ctx.channel.id}'
+        res = self.connection.select_records_by_id(channel_id=channel_id)
+        if res:
+            webhook_id = res[0]['webhook_id']
+            try:
+                webhook = await self.bot.fetch_webhook(webhook_id)
+                await webhook.delete()
+            except NotFound:
+                self.logger.warning("Could not find and delete previously stored webhook with id %s", webhook_id)
+            new_webhook = await ctx.channel.create_webhook(name=webhook_name)
+            self.connection.update_webhook_id_of_record(channel_id=channel_id, webhook_id=new_webhook.id)
+            await ctx.send(f"""Updated webhook with name {webhook_name}.""")
+        else:
+            await ctx.send(f"""Text channel has not been setup with a webhook.
+                            Please run z!tc setup <name-of-webhook> to start using the z!tc bot.""")
+
+    @tc.command()
+    async def clear(self, ctx):
+        """
+        Clears the text channel of the webhook created by the twitcasting bot. Deletes any subscriptions added in this text channel.
+            Usage: z!tc clear
+        """
+        channel_id = f'{ctx.guild.id}:{ctx.channel.id}'
+        res = self.connection.select_records_by_id(channel_id=channel_id)
+        if res:
+            webhook_id = res[0]['webhook_id']
+            try:
+                self.connection.delete_all_records(channel_id=channel_id)
+                webhook = await self.bot.fetch_webhook(webhook_id)
+                await webhook.delete()
+                await ctx.send(f"""Cleared text channel of twitcasting subscriptions and webhook.""")
+            except NotFound:
+                self.logger.warning("Could not find and delete previously stored webhook with id %s", webhook_id)
+        else:
+            await ctx.send(f"""Text channel has not been setup with a webhook.
+                            Could not clear the text channel of any bot artifacts (if any).""")
+
+    @tc.command(hidden=True)
+    async def update(self, ctx):
+        channel_id = f'{ctx.guild.id}:{ctx.channel.id}'
+        users = self.connection.select_records_by_id(channel_id=channel_id)
+        for user in users:
+            if user['user_id']:
+                new_user_info = self.twitcasting.search_users(words=user['user_id'], limit=1)
+                self.connection.update_name_of_record(user_id=user['user_id'], name=new_user_info[0].screen_id)
+
+    async def broadcast(self, request):
```

Comment:
docstring? this is the callback to discord I imagine this is where we'd end up making changes in the future